third_party/blink/web_tests/external/wpt/resources/chromium/cobalt/fake-h5vcc-system-impl.js

```diff
@@ -0,0 +1,56 @@
+import {H5vccSystem, H5vccSystemReceiver} from '/gen/cobalt/browser/h5vcc_system/public/mojom/h5vcc_system.mojom.m.js';
+
+// Implementation of h5vcc_system.mojom.H5vccSystem.
+class FakeH5vccSystemImpl {
+  constructor() {
+    console.log('FakeH5vccSystemImpl constructor');
+    this.interceptor_ =
+        new MojoInterfaceInterceptor(H5vccSystem.$interfaceName);
+    this.interceptor_.oninterfacerequest = e => this.bind(e.handle);
+    this.receiver_ = new H5vccSystemReceiver(this);
+    this.stub_result_ = '';
+  }
+
+  start() {
+    console.log('FakeH5vccSystemImpl start');
+    this.interceptor_.start();
+  }
+
+  stop() {
+    console.log('FakeH5vccSystemImpl stop');
+    this.interceptor_.stop();
+  }
+
+  reset() {
+    this.stub_result_ = '';
+  }
+
+  // getAdvertisingId() {
+  //   console.log('FakeH5vccSystemImpl getAdvertisingId returning Promise');
+  //   const result = this.stub_result_;
+  //   return new Promise((resolve) => {
+  //     console.log('Promise resolve() 111');
+  //     resolve(result);
+  //     console.log('resolve() is called');
+  //   });
+  // }
+
+  getAdvertisingId() {
+    console.log('async getAdvertisingId() return ' + this.stub_result_);
+    return this.stub_result_;
```

Comment:
Found the problem: this should be:
- A `return Promise.resolve( something )`;
- The `something` needs to be a dictionary and the key(s) need to have the same names as the mojom fields; in this case the mojom file looks like:
`GetAdvertisingId() => (string id);`
  so we should return here `{id: 'blabla'}`.

Putting it all together:
```
    return Promise.resolve({ id: this.stub_result_});
```

makes the 
```
    let result = await window.h5vcc.system.getAdvertisingId();
```
work.